modules/cloudsql-instance/variables.tf

```diff
@@ -169,9 +148,26 @@ variable "name" {
   type        = string
 }
 
-variable "network" {
-  description = "VPC self link where the instances will be deployed. Private Service Networking must be enabled and configured in this VPC."
-  type        = string
+variable "network_config" {
+  description = "Network configuration for the instance. Only one between private VPC and PSC can be used."
+  type = object({
+    authorized_networks = optional(map(string))
+    ipv4_enabled        = optional(bool, false)
+    private_network     = optional(string, null)
+    require_ssl         = optional(bool)
+    allocated_ip_ranges = optional(object({
+      primary = optional(string)
+      replica = optional(string)
+    }))
+    psc_config = optional(object({
+      allowed_consumer_projects = optional(list(string), [])
+      psc_enabled               = optional(bool, true)
+    }))
+  })
+  validation {
+    condition     = (var.network_config.private_network != null ? 1 : 0) + (var.network_config.psc_config != null ? 1 : 0) == 1
```

Comment:
This prevents having instance with no private network connectivity (which is rare, but still valid use case) - i.e. only public IP address. Also, this should be reflected in description.

Reply:
You're right, I fixed this.

Reply:
You're righ, I fixed this.